pkg/controller/dataimportcron-controller.go

```diff
@@ -557,22 +639,150 @@ func (r *DataImportCronReconciler) createImportDataVolume(ctx context.Context, d
+	dataSourceName := dataImportCron.Spec.ManagedDataSource
+	digest := dataImportCron.Annotations[AnnSourceDesiredDigest]
+	if digest == "" {
+		return nil
+	}
+	dvName, err := createDvName(dataSourceName, digest)
+	if err != nil {
+		return err
+	}
+
+	className, err := cc.GetSnapshotClassForSmartClone(dataVolume.Name, &dvStorageClass.Name, r.log, r.client)
+	if err != nil {
+		return err
+	}
+	labels := map[string]string{
+		common.CDILabelKey:       common.CDILabelValue,
+		common.CDIComponentLabel: "",
+	}
+	desiredSnapshot := &snapshotv1.VolumeSnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dvName,
+			Namespace: dataImportCron.Namespace,
+			Labels:    labels,
+		},
+		Spec: snapshotv1.VolumeSnapshotSpec{
+			Source: snapshotv1.VolumeSnapshotSource{
+				PersistentVolumeClaimName: &dvName,
+			},
+			VolumeSnapshotClassName: &className,
+		},
+	}
+	r.setDataImportCronResourceLabels(dataImportCron, desiredSnapshot)
+
+	currentSnapshot := &snapshotv1.VolumeSnapshot{}
+	if err := r.client.Get(context.TODO(), client.ObjectKeyFromObject(desiredSnapshot), currentSnapshot); err != nil {
 		if !k8serrors.IsNotFound(err) {
 			return err
 		}
-		dv := r.newSourceDataVolume(dataImportCron, dvName)
-		if err := r.client.Create(ctx, dv); err != nil && !k8serrors.IsAlreadyExists(err) {
+		cc.AddAnnotation(desiredSnapshot, AnnLastUseTime, time.Now().UTC().Format(time.RFC3339Nano))
+		if err := r.client.Create(ctx, desiredSnapshot); err != nil {
 			return err
 		}
 	} else {
-		if err := r.updatePvc(ctx, dataImportCron, pvc); err != nil {
-			return err
+		if currentSnapshot.Status != nil && currentSnapshot.Status.ReadyToUse != nil && *currentSnapshot.Status.ReadyToUse {
+			// Clean up PVC as that is not needed any more
+			pvc := &corev1.PersistentVolumeClaim{ObjectMeta: metav1.ObjectMeta{Name: desiredSnapshot.Name, Namespace: desiredSnapshot.Namespace}}
+			if err := r.client.Delete(ctx, pvc); err != nil && !k8serrors.IsNotFound(err) {
+				return err
+			}
 		}
 	}
-	dataImportCron.Status.CurrentImports = []cdiv1.ImportStatus{{DataVolumeName: dvName, Digest: digest}}
+
+	return nil
+}
+
+func (r *DataImportCronReconciler) updateDataImportCronSuccessCondition(ctx context.Context, dataImportCron *cdiv1.DataImportCron, format cdiv1.DataImportCronSourceFormat, pvcName string) error {
+	switch format {
+	case cdiv1.DataImportCronSourceFormatPvc:
+		updateDataImportCronCondition(dataImportCron, cdiv1.DataImportCronUpToDate, corev1.ConditionTrue, "Latest import is up to date", upToDate)
+	case cdiv1.DataImportCronSourceFormatSnapshot:
+		snapshot := &snapshotv1.VolumeSnapshot{}
+		err := r.client.Get(ctx, types.NamespacedName{Namespace: dataImportCron.Namespace, Name: pvcName}, snapshot)
+		if err != nil {
+			if !k8serrors.IsNotFound(err) {
+				return err
+			}
+			// Snapshot creation will trigger reconcile
+			return nil
+		}
```

Comment:
because this updates the status I think it makes more sense that it comes to conclusions on it's own based on it's own observation of the cluster

Reply:
I don't think that getting it again is a good practice. Even in the much larger DV controller we are getting all the CRs once in `sync()`, and once in `updateStatus()`.

Reply:
Yes, I am getting it in the "updateStatus" portion of the data import cron controller

Reply:
@awels Hey

We were hoping to get your thoughts on this one as the tiebreaker

Reply:
I am with Arnon on this one. If the status changed between us looking up the value earlier, and updating the status. Then that should trigger a new reconcile since we are watching the snapshotvolumes, and the next reconcile will properly update the condition to what it should be.